Adding user statistics

Cort98 opened the pull request:
- Created user controller and show action
- Created view for user profile
- Profile page shows Gravatar, along with total games, wins and losses
- Edited static root page to show different options if signed in or not

We can add number of forfeited matches later, but we would have to change its type in the database to hold the id of the user that forfeited

Cort98 committed:
Adding user statistics

*Created user controller and show action
*Created view for user profile
*Profile page shows Gravatar, along with total games, wins and losses
*Edited static root page to show different options if signed in or not

Cort98 committed:
Merge branch 'master' into user_scores

Conflicts:
	app/views/games/show.html.erb
	config/routes.rb

Cort98 committed:
small fix

Cort98 committed:
rubocop fix

Cort98 committed:
Added field to game to store id of forfeiting user

* show number of forfeited games
* show number of games played as white or black

mscoutermarsh commented on spec/helpers/users_helper_spec.rb:
```diff
@@ -0,0 +1,14 @@
+#   end
+# end
+RSpec.describe UsersHelper, type: :helper do
+end
```
Could remove this file since empty

mscoutermarsh commented:
Looks great, Cort. Go ahead and merge when you can 😄

Cort98 committed:
Added way to get winner through checkmate method.
Edited some tests to have users and to check for the winners id

Merge branch 'master' into user_scores

Conflicts:
	app/views/static_pages/index.html.erb

Cort98 committed:
Refactoring for rubocop

Cort98 committed:
Remove users_helper in spec/helpers
